NewHaven/src/Player.cpp

```diff
@@ -226,4 +226,12 @@ void Player::setVillage(VGMap v_map) {
 
 string Player::getID(){
     return *id;
-}
\ No newline at end of file
+}
+
+ResourceTracker Player::getScore(){
```

Comment:
getScore and setScore are unneccesary. resourceTracker() method already exposes those data fields for manipulation. This includes consuming resources and checking if enough resources are available. So you would want to assign the pointer of the turns player score to the next player. Make sure to save the original pointer as you will need to restore afterwards. You can also make a copy by using the assignment operator.

`*next_player.resourceTracker()->score = *current_turn_player.resourceTracker()->score;   `

Then you just do the reverse afterwards to update the resource score of the main player;

Reply:
Also, just a small note here. Another possibilty is to move the resourceScore to be held by the GameController and as such modify the player class to ask require a ResourceTracker to be passed when buildingVillage or placignHarvestTIle. If we do this than we can just have a prompt asking the other players if they want to build and use the same ResourceTracker for everyone. I think this would be the more efficient approach as realistically the player does not need their own game_board scoring facility.